src/sif/dynamiccost.cc

```diff
@@ -213,7 +222,16 @@ void DynamicCost::AddUserAvoidEdges(const std::vector<AvoidEdge>& avoid_edges) {
 
 Cost DynamicCost::BSSCost() const {
   return kNoCost;
-};
+}
+
+void DynamicCost::set_use_tracks(float use_tracks) {
+  // Calculate factor value based on use preference. Return value
+  // in range [kMaxTrackFactor; 1], if use < 0.5; or
+  // in range [1; kMinTrackFactor], if use > 0.5
+  track_factor_ = use_tracks < 0.5f
+                      ? (kMaxTrackFactor - 2.f * use_tracks * (kMaxTrackFactor - 1.f))
+                      : (kMinTrackFactor + 2.f * (1.f - use_tracks) * (1.f - kMinTrackFactor));
```

Comment:
ok cool so this works out to `.5` being a factor of `1`, which is good. then costings like auto (and probably other motor vehicles) have a default `use_tracks` that is a lower number. makes sense to me!